packages/hash/frontend/src/components/layout/PageHeader/ActionsDropdown.tsx

```diff
@@ -27,28 +27,31 @@ export const ActionsDropdown: React.FC<{
   const buttonRef = useRef(null);
 
   const [open, setOpen] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const { createUntitledPage } = useCreatePage(accountId);
 
   const id = open ? "actions-popover" : undefined;
 
-  const [createPageOpen, setCreatePageOpen] = useState(false);
-
-  const closeCreatePage = useCallback(() => {
-    // Prevent the bug of closing a non-existing modal
-    if (createPageOpen) {
-      setCreatePageOpen(false);
+  // @todo handle loading/error states properly
+  const addPage = useCallback(async () => {
+    if (loading) {
+      return;
     }
-  }, [createPageOpen]);
-
-  const newEntityTypeRoute = `/${accountId}/types/new`;
 
-  const showCreatePage = () => {
-    setCreatePageOpen(true);
-    if (open) {
+    setLoading(true);
+    try {
+      await createUntitledPage();
       setOpen(false);
+    } catch (err) {
+      // eslint-disable-next-line no-console -- TODO: consider using logger
+      console.error("Could not create page: ", err);
+      setLoading(false);
```

Comment:
Fixed here => https://github.com/hashintel/hash/pull/409/commits/a2c54ce0bd5494d78dfa3935a2647fc8fa5fe0d6. Leaving it there because I assume we'd want to handle the error differently in both places (ActionsDropdown and Sidebar). That can be properly decided when we get to working on that

Reply:
Doesn't have to be addressed in this PR, but I don't like the `try catch` method for catching GQL errors, and much prefer the declarative approach where a hook returns either a response or an error object which is correctly typed.

I think it might be worth considering restructuring the `useCreatePage` to be more consistent with the hooks it is abstracting/replacing (i.e. `useMutation`).

Also having re-read the hook now, I'm not sure whether `createPageFn` will throw an error? Isn't the error returned by the hook? i.e.

```TypeScript
const [setParentPageFn, { data, error }] = useMutation<...>(...)
```

Reply:
Yeah I agree, the reason why I didn't go with that approach was because I chain mutations in `createSubPage` and each mutation has it's error state. But I agree it might make more sense to split the hook into two hooks one to handle creating a new page and the other to handle creating sub/child pages. 

Yeah `createPageFn` will throw an error if you don't explicitly add the onError callback. 

I've added a comment to refactor the hook here => [e042d61](https://github.com/hashintel/hash/pull/409/commits/e042d6194b793297acf1aa3e6f8a7437d1f78ca2)

Reply:
Splitting the hook out into two would make sense to me, it'd let us interact with the mutation hooks in a consistent way throughout the repo. We could also support a `redirectToPage` boolean option for the method incase there are occasions where the redirect is undesired. Thanks for leaving the comment 👍